features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapReceiverImpl.java

```diff
@@ -106,7 +106,11 @@ public void setTrapdConfig(TrapdConfiguration newTrapdConfig) {
      */
     public static boolean compareSnmpV3UsersMap(Map<String, SnmpV3User> existingSnmpV3UserMap, Map<String, SnmpV3User> updatedSnmpV3Usermap) {
 
-        if (existingSnmpV3UserMap.isEmpty() || updatedSnmpV3Usermap.isEmpty()) {
+        if ((existingSnmpV3UserMap.isEmpty() && updatedSnmpV3Usermap.isEmpty())
+                || (!existingSnmpV3UserMap.isEmpty() && updatedSnmpV3Usermap.isEmpty())) {
```

Comment:
I also find this code to be confusing, why don't just implement a proper equals method in SnmpV3User instead?